src/config/cloudtrail.json

```diff
@@ -0,0 +1,753 @@
+{
+  "name": "CLOUD_TRAIL",
+  "displayName": "Cloud Trail Management Events / Logs",
+  "rules": [
+    {
+      "aws": {
+        "logContentParseAs": "json"
+      },
+      "sources": [
+        {
+          "sourceType": "logs",
+          "source": "log_group",
+          "condition": "$prefix('aws-cloudtrail-logs')"
+        }
+      ],
+      "attributes": [
+        {
+          "key": "aws.service",
+          "pattern": "'cloudtrail'"
+        },
+        {
+          "key": "audit.action",
+          "pattern": "log_content_parsed.eventName"
+        },
+        {
+          "key": "audit.identity",
+          "pattern": "log_content_parsed.userIdentity.arn"
+        },
+        {
+          "key": "audit.result",
+          "description": "If a field errorCode exists, then return Failure.{errorCode}, else return Succeeded",
+          "pattern": "if( log_content_parsed.errorCode==null, &'Succeeded', &[`Failed`, log_content_parsed.errorCode] | [? @!=null ] | join('.', @) , @)"
```

Comment:
Is `[? @!=null ]` neede here? We already checked that `log_content_parsed.errorCode` is not null

Reply:
log_content_parsed.errorCode = null
[`Failed`, log_content_parsed.errorCode] | [? @!=null ] | join('.', @) > "Failed"
[`Failed`, log_content_parsed.errorCode] | join('.', @) > crash, TypeError: join() expected argument 2 to be type array-string but received type array instead.

For jmespath string and null (and so the arrays thereof) are different types but they are not both allowed in most jmespath methods definitions (only string). I could overwrite join the way I ended up overwriting built-in starts_with if you want - I just did both things differently at different stages.

Reply:
Btw, just to have it said out loud, I decided to translate
{  "log_content_parsed": 
    {"errorCode": null} }
as Failed (same for empty string, even though there is no errorCode. There is another field of errorMessage, that could still be not empty. As a user, I'd rather have an extra false positive Failed message, that I could clarify by looking at the log, rather than have a Failed hide somewhere in the ocean of Succeeded. 

I could check on errorMessage field, too, but that seems an edge case that didn't deserve expanding the pattern.

Btw, one more thing
log_content_parsed.errorCode = ""
[`Failed`, log_content_parsed.errorCode] | [? @!=null ] | join('.', @) > "Failed."

but we could potentially filter that out if you want:
[`Failed`, log_content_parsed.errorCode] | [? @!=null && @!=``] | join('.', @) > "Failed"

Btw, Azure and GCP also has Succeeded/Failed.somethingMore. Not sure if I checked the cases above against them.

Reply:
```
log_content_parsed.errorCode = null
[Failed, log_content_parsed.errorCode] | [? @!=null ] | join('.', @) > "Failed"
[Failed, log_content_parsed.errorCode] | join('.', @) > crash, TypeError: join() expected argument 2 to be type array-string but received type array instead.
```
Will this scenario ever happen though? The expression should evaluate the on-true argument here (`&'Succeeded'`) and not evaluate the on-false argument. Or is my assumption about short-circuit evaluation wrong in JMESPath?